docs/authentication/auth-code-pkce-flow.md

```diff
@@ -0,0 +1,160 @@
+# Authorization Code Flow with Proof Key for Code Exchange
+
+The "Authorization Code Flow with Proof Key for Code Exchange" also simply "Auth code flow with PKCE" is the most recommended form of authenticating RingCentral users and exchange tokens in client side applications. It is an enhancement of [Authorization Code Flow](./auth_code_flow):
+
+* No client secret required in client side.
+* Exchange token with code and code verifier.
+
+The general flow looks like this:
+
+<img src="../../img/auth_code_pkce_flow.png" class="img-fluid">
+
+The step-by-step details of this flow are explained below.
+
+## Step 1. Generate code verifier and code challenge
+
+Before we open an authorization request to RingCentral API server, we need to generate a pair of code verifier and code challenge.
+
+Code verifier should be a cryptographically-random string, without '+', '/', and '=' characters.
+
+**Example Code Verifier**
+
+a cryptographically-random example string:
+
+```
+pIUgx4tiqFpaOUz0HMc_QbIyQlL901w8mRmkrmhEJ_E
+```
+
+Code challenge is generated from code challenge. For devices that can perform a SHA256 hash, the code challenge is a BASE64-URL-encoded string of the SHA256 hash of the code verifier. Clients that do not have the ability to perform a SHA256 hash are permitted to use the plain code verifier string as the challenge.
+
+**Example Code Challenge**
+
+Generated from upper code verifier with SHA256 hash and BASE64-URL-encoded:
+
+```
+_drLS7o5FwkfUiBhlq2hwJnK_SC6yE7sKOde5O1fdzk
+```
+
+## Step 2. Request authorization code
+
+When your application needs to access a user's data, redirect the user to the RingCentral API server. The authorization URL is same as URL from [Authorization Code Flow Step 1](./auth_code_flow#step-1-request-authorization-code), and PKCE flow will need additional parameters `code_challenge` and `code_challenge_method` in authorization URL:
+
+| Parameter       | Type   | Description |
+| --------------- | ------ | ----------- |
+| `code_challenge` | string | Required. Generated from code challenge
```

Comment:
"code_challenge string Required. Generated from code challenge <- this should say code verifier

Reply:
It should be `code_challenge`. We send `code_challenge` to server by authorization URI. And send `code_verifier`  to server only  in token exchange api.